### PR TITLE
refactor: rename structures related to Modules

### PR DESCRIPTION
**Pull request by bartlomieju**
* rename structures related to ES Modules; add "Modules" prefix
* remove unneeded `Unpin` trait requirement for "ModuleLoader"

### CODE REVIEW

File: core/modules.rs [hunk 1]
```diff
@@ -20,10 +19,31 @@ use std::rc::Rc;
...
+///
+/// Users can implement `Into<ModuleInfo>` for different file types that
+/// can be transpiled to valid EsModule.
+///
+/// Found module URL might be different from specified URL
+/// used for loading due to redirections (like HTTP 303).
+/// Eg. Both "https://example.com/a.ts" and
+/// "https://example.com/b.ts" may point to "https://example.com/c.ts"
+/// By keeping track of specified and found URL we can alias modules and avoid
+/// recompiling the same code 3 times.
```
Comment (ry) [inline core/modules.rs]: Why 3?

Comment (bartlomieju) [reply core/modules.rs]: Don't know; old comment, PTAL at my TODO comment below

Comment (ry) [reply core/modules.rs]: Having some bookkeeping of redirects sounds useful.

Comment (bartlomieju) [reply core/modules.rs]: CC @piscisaureus

**Pull request merged**